app/models/mixins/per_ems_worker_mixin.rb

```diff
@@ -65,7 +65,8 @@ def start_workers
 
     def start_worker_for_ems(ems_or_queue_name)
       params = {:queue_name => queue_name_for_ems(ems_or_queue_name)}
-      start_worker(params)
+      ems = ems_or_queue_name.kind_of?(ExtManagementSystem) ? ems_or_queue_name : ems_from_queue_name(ems_or_queue_name)
+      start_worker(params) unless ems.disabled
     end
```

Comment:
I'm half-guesing but I thing it's better to do the check in `sync_workers` where it computes `desired`, similar to how it checks `has_required_role?`.  That way disabling would also cause `stop_worker_for_ems()`.
Also probably add `disabled` to `return unless self.has_required_role?` in start_workers.